File: vbnet/word_reverser/word_reverser_5.vb
' This program is called Word Reverser.
' The purpose of this program is to reverse the characters in each word of a given sentence.
' It takes a sentence as input from the user, processes each word, and outputs the sentence with each word reversed.
' The program demonstrates basic string manipulation and array handling in Visual Basic .NET.

Module WordReverser

    Sub Main()
        ' Prompt the user to enter a sentence
        Console.WriteLine("Please enter a sentence:")

        ' Read the input sentence from the console
        Dim inputSentence As String = Console.ReadLine()

        ' Split the sentence into words using space as the delimiter
        Dim words() As String = inputSentence.Split(" "c)

        ' Initialize an empty list to hold the reversed words
        Dim reversedWords As New List(Of String)

        ' Loop through each word in the array
        For i As Integer = 0 To words.Length - 1
            ' Reverse the characters in the word
            Dim reversedWord As String = ReverseString(words(i))

            ' Add the reversed word to the list
            reversedWords.Add(reversedWord)
        Next

        ' Join the reversed words into a single sentence
        Dim reversedSentence As String = String.Join(" ", reversedWords)

        ' Output the reversed sentence to the console
        Console.WriteLine("Reversed sentence: " & reversedSentence)

        ' Wait for the user to press a key before closing the console window
        Console.ReadKey()
    End Sub

    ' Function to reverse the characters in a string
    Function ReverseString(ByVal str As String) As String
        ' Convert the string to a character array
        Dim charArray() As Char = str.ToCharArray()

        ' Initialize an empty string to hold the reversed characters
        Dim reversedStr As String = String.Empty

        ' Loop through the character array in reverse order
        For i As Integer = charArray.Length - 1 To 0 Step -1
            ' Append each character to the reversed string
            reversedStr &= charArray(i)
        Next

        ' Return the reversed string
        Return reversedStr
    End Function

End Module

File: python/word_reverser/word_reverser_5.py
# This program is called "Word Reverser". The purpose of this program is to take a sentence as input
# and reverse each word in the sentence while maintaining the original order of the words.
# The program will prompt the user to enter a sentence, then it will process the sentence and output
# the reversed words. This can be useful for various text processing tasks and helps in understanding
# string manipulation in Python.

def reverse_word(word):
    # This function takes a single word as input and returns the reversed version of the word.
    # It uses slicing to reverse the string.
    return word[::-1]

def reverse_sentence(sentence):
    # This function takes a sentence as input and returns a new sentence with each word reversed.
    # It splits the sentence into words, reverses each word, and then joins them back together.
    words = sentence.split()  # Split the sentence into words
    reversed_words = []
    for word in words:
        reversed_words.append(reverse_word(word))  # Reverse each word and add to the list
    return ' '.join(reversed_words)  # Join the reversed words back into a sentence

def main():
    # The main function of the program. It prompts the user to enter a sentence,
    # calls the reverse_sentence function to process the input, and then prints the result.
    user_input = input("Enter a sentence to reverse each word: ")  # Prompt the user for input
    result = reverse_sentence(user_input)  # Reverse the words in the input sentence
    print("Reversed sentence:", result)  # Print the result

if __name__ == "__main__":
    main()  # Call the main function to start the program

